sklearn/isotonic.py

```diff
@@ -5,10 +5,72 @@
 
 import numpy as np
 from scipy import interpolate
+from scipy.stats import spearmanr
 from .base import BaseEstimator, TransformerMixin, RegressorMixin
 from .utils import as_float_array, check_arrays
 from ._isotonic import _isotonic_regression
 import warnings
+import math
+
+
+def check_increasing(x, y):
+    """Determine whether the relationship between x and y appears to be
+    increasing or decreasing based on Spearman correlation.
+
+    Parameters
+    ----------
+    x : array-like, shape=(n_samples,)
+            Training data.
+
+    y : array-like, shape=(n_samples,)
+        Training target.
+
+    Returns
+    -------
+    `increasing_bool` : boolean
+        Whether the relationship is increasing or decreasing.
+
+    Notes
+    -----
+    Determine whether the relationship between x and y appears to be
+    increasing or decreasing.  The Spearman correlation coefficient is
+    estimated from the data, and the sign of the resulting estimate
+    is used as the result.
+
+    In the event that the 95% confidence interval based on Fisher transform
+    spans zero, a warning is raised.
+
+    References
+    ----------
+    Fisher transformation. Wikipedia.
+    http://en.wikipedia.org/w/index.php?title=Fisher_transformation
+    """
+
+    # Calculate Spearman rho estimate and set return accordingly.
+    rho, _ = spearmanr(x, y)
+    if rho >= 0:
+        increasing_bool = True
+    else:
+        increasing_bool = False
```

Comment:
@ogrisel, if you check below, you can see that the user provides an input `increasing` which may be either string or True/False.  If a string is provided, the proper method is applied to determine the true direction, thereby setting `increasing_bool`.  Agreed on other comments but just want to make sure we see the reason re: `increasing` vs. `increasing_bool`